index.js

```diff
@@ -56,8 +57,13 @@ Parser.prototype._transform = function (data, enc, cb) {
   }
 
   for (var i = start; i < buf.length; i++) {
-    if (buf[i] === this.escape) this._escaped = !this._escaped
-    if (!this._escaped) {
+    // non-escaped quote (quoting the cell)
+    if (buf[i] === this.quote && buf[i - 1] !== this.escape) {
```

Comment:
could you check that `i > 0` after the `&&`? otherwise we're gonna get a deopt from v8